source/develop/release-management/features/integration/centos-9-stream-support.md

```diff
@@ -0,0 +1,100 @@
+---
+title: CentOS Stream 9 support
+category: feature
+authors:
+  - sandrobonazzola
+---
+
+# {{ page.title }}
+
+## Summary
+
+Add support for CentOS Stream 9 to oVirt project
+
+## Owner
+
+{% assign owner = site.data.authors["sandrobonazzola"] %}
+{% if owner.github %}
+* Name: [{{ owner.name }}](https://github.com/{{ owner.github }})
+{% else %}
+* Name: {{ owner.name }}
+{% endif %}
+{% if owner.email %}
+* Email: <{{owner.email}}>
+{% endif %}
+
+
+## Current Status
+- Last updated: Tue Jul 27 2021
+- Target release: oVirt 4.5.0
+- Reference bug: Bug [1986335](https://bugzilla.redhat.com/1986335) - [RFE] Support CentOS Stream 9
+
+
+## Detailed Description
+
+- Support building oVirt project packages on CentOS Stream 9
+- Add automation for building and testing oVirt project packages on CentOS Stream 9
+
+## Prerequisites
+
+- [mock-core-configs-34.5-1](https://github.com/rpm-software-management/mock/releases/tag/mock-core-configs-34.5-1) is needed in order to get CentOS Stream 9 build environment.
+
+## Limitations
+
+There are currently no known limitations.
+
+## Benefit to oVirt
+
+oVirt will be able to run on top of CentOS Stream 9 and RHEL 9 once it will be available.
+
+## Entity Description
+
+No new entity needed in oVirt Engine.
```

Comment:
no new entity, sure
but what about emulated machines? we still use the RHEL 8 ones because we cannot introduce a cluster level with RHEL 9 stuff, right?
and I guess emulated machines of RHEL 7 are dropped in RHEL 9, right? so VDSM for CentOS Stream 9 or RHEL 9 won't support compatibility levels 4.2 and 4.3?

Reply:
Well, if we are really going to introduce RHEL 9 hosts, then I think they should be in their own cluster level. Meaning that RHEL 9 hosts could be added only to latest cluster levels 4.7 (RHEL 8.5) or 4.8 (RHEL 8.6) and then we should introduce new cluster level (either 5.0 or 9.0) where only RHEL 9 hosts will be supported

Reply:
At current state, RHEL 7 machines are still supported in CentOS Stream 9:

```bash
# rpm -qf /usr/libexec/qemu-kvm
qemu-kvm-core-6.0.0-10.el9.x86_64

# /usr/libexec/qemu-kvm -machine help
Supported machines are:
pc                   RHEL 7.6.0 PC (i440FX + PIIX, 1996) (alias of pc-i440fx-rhel7.6.0)
pc-i440fx-rhel7.6.0  RHEL 7.6.0 PC (i440FX + PIIX, 1996) (default)
pc-i440fx-rhel7.5.0  RHEL 7.5.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.4.0  RHEL 7.4.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.3.0  RHEL 7.3.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.2.0  RHEL 7.2.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.1.0  RHEL 7.1.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.0.0  RHEL 7.0.0 PC (i440FX + PIIX, 1996)
q35                  RHEL-8.5.0 PC (Q35 + ICH9, 2009) (alias of pc-q35-rhel8.5.0)
pc-q35-rhel8.5.0     RHEL-8.5.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel8.4.0     RHEL-8.4.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel8.3.0     RHEL-8.3.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel8.2.0     RHEL-8.2.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel8.1.0     RHEL-8.1.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel8.0.0     RHEL-8.0.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.6.0     RHEL-7.6.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.5.0     RHEL-7.5.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.4.0     RHEL-7.4.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.3.0     RHEL-7.3.0 PC (Q35 + ICH9, 2009)
none                 empty machine
```

Reply:
> Well, if we are really going to introduce RHEL 9 hosts, then I think they should be in their own cluster level. Meaning that RHEL 9 hosts could be added only to latest cluster levels 4.7 (RHEL 8.5) or 4.8 (RHEL 8.6) and then we should introduce new cluster level (either 5.0 or 9.0) where only RHEL 9 hosts will be supported

can you help covering the entities needed for adding the new cluster level?

Reply:
re machine types
- the fact that they are there now doesn't mean they're going to stay. please confirm qemu plans, @ahadas .
- we do not necessarily need to change them. CL shouldn't be used to define host version support. It's for VM HW/capabilities. I wouldn't complicate it from the start, we can keep everything in place (or drop <4.4 CL in el9 vdsm if those types are going away after all)

Reply:
>     * the fact that they are there now doesn't mean they're going to stay. please confirm qemu plans, @ahadas .

They will not be supported

Reply:
Added notes about this discussion in the last commit.